### PR TITLE
Blaze Optimization: Remove banner on product list screen

### LINKED ISSUE #10954: Blaze Optimization: Remove banner on product list screen

### PR DESCRIPTION
**Pull request by hafizrahman**
Closes: #10954

## Description

This PR removes the Blaze banner being displayed on the Products list screen.

## Testing instructions

Start a fresh build and go to Products list screen, ensure no Blaze banner is shown on top. A quick read through the codebase should suffice, too.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift [hunk 1]
```diff
@@ -12,20 +12,15 @@ class ProductListViewModel {
     let siteID: Int64
     private let stores: StoresManager
     private let userDefaults: UserDefaults
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift]: We can remove the `userDefaults` as we no longer use it.

File: WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift [hunk 2]
```diff
@@ -161,25 +156,7 @@ class ProductListViewModel {
...
-        async let storeHasPublishedProducts = (try? checkIfStoreHasProducts(siteID: siteID, status: .published)) ?? false
-        async let storeHasAnyOrders = (try? checkIfStoreHasOrders(siteID: siteID)) ?? false
-        guard await(isSiteEligible, storeHasPublishedProducts, storeHasAnyOrders) == (true, true, false) else {
-            return false
-        }
-        return !userDefaults.hasDismissedBlazeBanner(for: siteID)
-    }
-
     @MainActor
     private func checkIfStoreHasProducts(siteID: Int64, status: ProductStatus? = nil) async throws -> Bool {
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift]: We can remove these `checkIfStoreHasProducts` and `checkIfStoreHasOrders` methods as we no longer use them.

File: WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift [hunk 3]
```diff
@@ -862,59 +858,6 @@ private extension ProductsViewController {
...
-    }
-
-    func updateBlazeBannerVisibility() {
-        Task { @MainActor in
-            await viewModel.updateBlazeBannerVisibility()
-        }
-    }
-
-    func showBlazeBanner() {
-        guard blazeBannerHostingController == nil else {
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift:888]: We can remove this variable as we don't use it anymore. 

```swift
private var blazeBannerHostingController: BlazeBannerHostingController?
```

**Review by selanthiraiyan — COMMENTED** [review]
Great job navigating `UIKit` code, Hafiz! 🎉 

I left a few minor comments about deleting unused code.

File: WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift (hunk 1 above)
Comment (hafizrahman) [reply WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift]: Thanks! Removed in https://github.com/woocommerce/woocommerce-ios/pull/10960/commits/33bd112d7e4a7abb2a869a95f57c6d622465ceb8

File: WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift (hunk 2 above)
Comment (hafizrahman) [reply WooCommerce/Classes/ViewRelated/Products/ProductsListViewModel.swift]: This is also removed in https://github.com/woocommerce/woocommerce-ios/pull/10960/commits/33bd112d7e4a7abb2a869a95f57c6d622465ceb8

File: WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift (hunk 3 above)
Comment (hafizrahman) [reply WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift:888]: Removed in https://github.com/woocommerce/woocommerce-ios/pull/10960/commits/1d317e12c23f7280d476608eac1af204e9c7f642

**Comment by hafizrahman** [comment]
Thanks for your eagle-eye check and finding those missing items to delete @selanthiraiyan . Also for the Slack discussion on ways to find unused elements. This should be good to re-review.

**Review by selanthiraiyan — APPROVED** [review]
Thanks for the updates, Hafiz! 🙇 🚀

**Pull request merged**